src/lsp-server.spec.ts

```diff
@@ -201,6 +201,70 @@ describe('completion', () => {
         server.didCloseTextDocument({ textDocument: doc });
     });
 
+    it('completions for clients that do not support insertReplaceSupport', async () => {
+        const doc = {
+            uri: uri('bar.ts'),
+            languageId: 'typescript',
+            version: 1,
+            text: `
+                class Foo {
+                    getById() {};
+                }
+
+                const foo = new Foo()
+                foo.getById()
+            `,
+        };
+        server.didOpenTextDocument({ textDocument: doc });
+        const proposals = await server.completion({ textDocument: doc, position: positionAfter(doc, '.get') });
+        assert.isNotNull(proposals);
+        const completion = proposals!.items.find(completion => completion.label === 'getById');
+        assert.isDefined(completion);
+        assert.isDefined(completion!.textEdit);
+        assert.containsAllKeys(completion!.textEdit, ['newText', 'range']);
+        server.didCloseTextDocument({ textDocument: doc });
+    });
+
+    it('completions for clients that support insertReplaceSupport', async () => {
+        const clientCapabilitiesOverride: lsp.ClientCapabilities = {
+            textDocument: {
+                completion: {
+                    completionItem: {
+                        insertReplaceSupport: true,
+                    },
+                },
+            },
+        };
+        const localServer = await createServer({
```

Comment:
One thing to note,
not sure if it is the best way to create a localServer here and to clean up at the end of this function.
This test is taking ~5s .

Reply:
It's what it is :)